tests/components/kostal_plenticore/test_config_flow.py

```diff
@@ -281,13 +352,73 @@ async def test_already_configured(hass: HomeAssistant) -> None:
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
 
-    result2 = await hass.config_entries.flow.async_configure(
+    result = await hass.config_entries.flow.async_configure(
         result["flow_id"],
         {
             "host": "1.1.1.1",
             "password": "test-password",
         },
     )
 
-    assert result2["type"] is FlowResultType.ABORT
-    assert result2["reason"] == "already_configured"
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_reconfigure(
```

Comment:
For the reconfigure flow we should also test the exceptions. These tests should also finish the flow (previous tests keep them until flowResultType.FORM, but ideally these also finish, but that would be out of the scope of this PR).

This way we test that the flow is able to recover properly